### PR TITLE
[core] Replace three maps/mutexes in GlyphAtlas with a single map and mutex

### PR DESCRIPTION
**Pull request by jfirebaugh**
Fixes the `GlyphAtlas` equivalent of #6835.

### CODE REVIEW

File: src/mbgl/text/glyph_atlas.cpp [hunk 1]
```diff
@@ -64,16 +65,8 @@ bool GlyphAtlas::hasGlyphRanges(const FontStack& fontStack, const GlyphRangeSet&
...
 util::exclusive<GlyphSet> GlyphAtlas::getGlyphSet(const FontStack& fontStack) {
-    auto lock = std::make_unique<std::lock_guard<std::mutex>>(glyphSetsMutex);
-
-    auto it = glyphSets.find(fontStack);
-    if (it == glyphSets.end()) {
-        it = glyphSets.emplace(fontStack, std::make_unique<GlyphSet>()).first;
-    }
-
-    // FIXME: We lock all GlyphSets, but what we should
-    // really do is lock only the one we are returning.
```
Comment (kkaefer) [inline src/mbgl/text/glyph_atlas.cpp]: This Fixme isn't resolved, is it?

Comment (jfirebaugh) [inline src/mbgl/text/glyph_atlas.cpp]: No, but I don't think it'll ever be fixed in the way described here. Instead, we'll do #667.

**Comment by tmpsantos** [comment]
This causes `platform/node/test/memory.test.js` to deadlock on `src/mbgl/text/glyph_pbf.cpp:120`.

**Comment by tmpsantos** [comment]
> This causes platform/node/test/memory.test.js to deadlock on src/mbgl/text/glyph_pbf.cpp:120.

After some investigation, the cause seems to be related to the mock file source used in that test returning the reply synchronously.

**Pull request merged**